src/lib/shady-render.ts

```diff
@@ -22,31 +23,83 @@ declare global {
   }
 }
 
+/**
+ * Template factory which scopes template DOM using ShadyCSS.
+ * @param scopeName {string}
+ */
 const shadyTemplateFactory = (scopeName: string) =>
     (result: TemplateResult) => {
-      const cacheKey = `${result.type}--${scopeName}`;
-      let templateCache = templateCaches.get(cacheKey);
-      if (templateCache === undefined) {
-        templateCache = new Map<TemplateStringsArray, Template>();
-        templateCaches.set(cacheKey, templateCache);
-      }
-      let template = templateCache.get(result.strings);
-      if (template === undefined) {
-        const element = result.getTemplateElement();
+  const cacheKey = `${result.type}--${scopeName}`;
+  let templateCache = templateCaches.get(cacheKey);
+  if (templateCache === undefined) {
+    templateCache = new Map<TemplateStringsArray, Template>();
+    templateCaches.set(cacheKey, templateCache);
+  }
+  let template = templateCache.get(result.strings);
+  if (template === undefined) {
+    const element = result.getTemplateElement();
+    window.ShadyCSS.prepareTemplateDom(element, scopeName);
+    template = new Template(result, element);
+    templateCache.set(result.strings, template);
+  }
+  return template;
+};
 
-        if (typeof window.ShadyCSS === 'object') {
-          window.ShadyCSS.prepareTemplate(element, scopeName);
-        }
+const TEMPLATE_TYPES = ['html', 'svg'];
+function removeStylesFromLitTemplates(scopeName: string) {
+  TEMPLATE_TYPES.forEach((type) => {
+    const templates = templateCaches.get(`${type}--${scopeName}`);
+    if (templates) {
+      templates.forEach((template) => {
+        const {element: {content}} = template;
+        const styles = content.querySelectorAll('style');
+        removeNodesFromTemplate(template, Array.from(styles));
+      });
+    }
+  });
+}
 
-        template = new Template(result, element);
-        templateCache.set(result.strings, template);
-      }
-      return template;
-    };
+const shadyRenderSet = new Set<string>();
+
+function hostForNode(node: Node) {
+  return node.nodeType === Node.DOCUMENT_FRAGMENT_NODE && (node as ShadowRoot).host
+}
 
 export function render(
     result: TemplateResult,
     container: Element|DocumentFragment,
     scopeName: string) {
-  return baseRender(result, container, shadyTemplateFactory(scopeName));
+  const host = hostForNode(container);
+  if (host && typeof window.ShadyCSS === 'object') {
+    const templateFactory = shadyTemplateFactory(scopeName);
+    const renderer = (container: Element|DocumentFragment, fragment: DocumentFragment) => {
```

Comment:
Talked offline, and it seems better to copy & paste the contents of render() for now, rather than expose an unstable API in the form of the renderDOM callback

Reply:
done